File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> ans;
        int m = nums1.size(), n = nums2.size();
        unordered_map<int,int> mp ;
        for(int val : nums1){
            mp[val]++;
        }
        for(int val : nums2){
            if(mp[val] >= 1){
                ans.push_back(val);
                mp[val] = 0;
            }
        }
        return ans;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        ArrayList<Integer> ans = new ArrayList<>();

        for(int val : nums1){
            mp.put(val, mp.getOrDefault(val,0)+1);
        }
        for(int val : nums2){
            if(mp.containsKey(val)){
                ans.add(val);
                mp.remove(val);
            }
        }
        return ans.stream().mapToInt(i -> i).toArray();
    }
}